src/Test/TestDamageStrategy.java

```diff
@@ -0,0 +1,138 @@
+import Units.SubModel;
+import Units.TCopterModel;
+import Units.TankModel;
+import Units.UnitModel;
+import Units.UnitModel.UnitEnum;
+import Units.Weapons.WeaponModel;
+import Units.Weapons.Damage.DamageStrategy;
+import Units.Weapons.Damage.StandardDamage;
+
+public class TestDamageStrategy extends TestCase
+{
+  private static DamageStrategy standardDamage = new StandardDamage();
+  private static ArrayList<UnitModel> unitModels;
+  private static ArrayList<WeaponModel> weaponModels;
+  private static int numUnitModels = UnitModel.UnitEnum.values().length;
+  private static int numWeaponModels = WeaponModel.WeaponType.values().length;
+
+  /** Make two COs and a GameMap to use with this test case. */
+  private void setupTest()
+  {
+    // TODO Obviously we don't want to hard-code the UnitModel array.
+    // order is INFANTRY, MECH, RECON, TANK, MD_TANK, NEOTANK, APC, ARTILLERY, ROCKETS, ANTI_AIR, MOBILESAM, FIGHTER, BOMBER, B_COPTER, T_COPTER, BATTLESHIP, CRUISER, LANDER, SUB
+    // this order is needed because that's how the unit and weaponmodel enums are currently set up.
+    unitModels = new ArrayList<UnitModel>(19);
+    unitModels.add(new InfantryModel());
+    unitModels.add(new MechModel());
+    unitModels.add(new ReconModel());
+    unitModels.add(new TankModel());
+    unitModels.add(new MDTankModel());
+    unitModels.add(new NeotankModel());
+    unitModels.add(new APCModel());
+    unitModels.add(new ArtilleryModel());
+    unitModels.add(new RocketsModel());
+    unitModels.add(new AntiAirModel());
+    unitModels.add(new MobileSAMModel());
+
+    unitModels.add(new FighterModel());
+    unitModels.add(new BomberModel());
+    unitModels.add(new BCopterModel());
+    unitModels.add(new TCopterModel());
+
+    unitModels.add(new BattleshipModel());
+    unitModels.add(new CruiserModel());
+    unitModels.add(new LanderModel());
+    unitModels.add(new SubModel());
+
+    // We build an array of weapons by taking each unit's weapons in turn.
+    weaponModels = new ArrayList<WeaponModel>();
+    for( UnitModel unit : unitModels )
+    {
+      if( null != unit.weaponModels )
+      {
+        for( WeaponModel weapon : unit.weaponModels )
+        {
+          weaponModels.add(weapon);
+        }
+      }
+    }
+
+  }
+
+  @Override
+  public boolean runTest()
+  {
+    setupTest();
+
+    boolean testPassed = true;
+    testPassed &= validate(true, "  Everything you know is wrong.");
+//    System.out.println(getDeltaString(new StandardDamage()));
+    return testPassed;
+  }
+
+  /**
+   * The values in the output array denote how much extra damage each interaction has with the new system compared to the standard. Output can be negative.
+   * @param other: which damage strategy to compare
+   * @return an array comprising the difference between the two damage strategy results
+   */
+  public int[][] getDeltas(DamageStrategy other)
+  {
+    int[][] deltas = new int[numWeaponModels][numUnitModels];
```

Comment:
So, you might be able to get away with this in Java because it's garbage-collected, but in e.g. C++ I'd be real nervous about creating a "new" array and then returning it as an argument, unless the function name clearly indicated is was allocating memory that it did not intend to free (like in a factory class, for example). Not sure I care too much here (because again, Java), but just something to be aware of.